### PR TITLE
Dailywork ftschindler

### PR DESCRIPTION
**Pull request by ftalbrecht**
Please have a look at the change in configuration, @renemilk. Otherwise I 've had problems in cases where we have command line overrides but no config file.
Would that break anything for you?

### CODE REVIEW

File: dune/xt/common/bisect.hh [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+#include <dune/xt/common/exceptions.hh>
+#include <dune/xt/common/float_cmp.hh>
+
+namespace Dune {
+namespace XT {
+namespace Common {
+
+
+double find_largest_by_bisection(const double& left, const double& right, std::function<double(const double&)> test)
```
[the remark this answers is not in the document]
Comment (renemilk) [reply dune/xt/common/bisect.hh]: Also, why does the test return double?

[the remark this answers is not in the document]
Comment (ftalbrecht) [reply dune/xt/common/bisect.hh]: The test should return a boolean, of course. And, as clearly documented :), this looks for the largest number between left and right, such that test is true, and the means to do so is bisection.
I will add docs and a test...

[the remark this answers is not in the document]
Comment (renemilk) [reply dune/xt/common/bisect.hh]: I see. Why can't 'left' be that number though? Also, I think that code can return an ```rr``` w/o ```test(rr) == true```.

**Comment by renemilk** [comment]
Apparently I'm not smart enough to handle the review buttons correctly, sorry.

**Comment by renemilk** [comment]
I still think that code can return an ```rr s.t. condition(rr) != true``` maybe I can come up with a failing test.

**Comment by renemilk** [comment]
I think the note in the docs precludes the failure I imagined. Or I imagined it entirely :wink: 
Anywho, I'm pushing a simplification for the exception test, but otherwise I'm happy.

File: dune/xt/common/test/bisection.cc [hunk 2]
```diff
@@ -28,13 +28,7 @@ GTEST_TEST(find_largest_by_bisection_test, works_for_right_smaller_left)
...
-  try {
-    find_largest_by_bisection(/*left=*/0, /*right=*/1, /*condition=*/[&](const auto& /*x*/) { return false; });
-  } catch (const Exceptions::bisection_error&) {
-    // everything in order
-    return;
-  } catch (...) {
-    EXPECT_TRUE(false) << "we should not get here";
-  }
-  EXPECT_TRUE(false) << "we should not get here";
+  EXPECT_THROW(
```
Comment (ftalbrecht) [inline dune/xt/common/test/bisection.cc]: nice one, thx

**Pull request merged**